src/frontend/test_runner/tests/testdata/agg.yaml

```diff
@@ -698,3 +698,20 @@
       LogicalAgg { group_key: [t.v1, flag], aggs: [] }
         LogicalExpand { column_subsets: [[], [t.v1]] }
           LogicalScan { table: t, columns: [t.v1] }
+- sql: |
+    /* input is sharded by group key */
+    create table t(x int);
+    create index i on t(x);
+    select count(*) as cnt from i group by x;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchProject { exprs: [count] }
+        BatchHashAgg { group_key: [i.x], aggs: [count] }
+          BatchScan { table: i, columns: [i.x], distribution: UpstreamHashShard(i.x) }
+  stream_plan: |
+    StreamMaterialize { columns: [cnt, i.x(hidden)], pk_columns: [i.x] }
+      StreamProject { exprs: [sum(count), i.x] }
+        StreamHashAgg { group_key: [i.x], aggs: [sum(count), sum(count)] }
+          StreamHashAgg { group_key: [i.x, Vnode(i.x)], aggs: [count, count] }
+            StreamProject { exprs: [i.x, i.t._row_id, Vnode(i.x)] }
+              StreamTableScan { table: i, columns: [i.x, i.t._row_id], pk: [i.t._row_id], distribution: UpstreamHashShard(i.x) }
```

Comment:
New test for hash agg without exchange. BTW, is the stream 2-phase agg needed?

Reply:
I think not, we can fix it later.